CRM/Contribute/Form/Contribution/Confirm.php

```diff
@@ -615,11 +615,7 @@ public function buildQuickForm() {
       // Recursively set defaults for nested fields
       if (isset($contact[$name]) && is_array($contact[$name]) && ($name == 'onbehalf' || $name == 'honor')) {
         foreach ($contact[$name] as $fieldName => $fieldValue) {
-          if (is_array($fieldValue) && !in_array($this->_fields[$name][$fieldName]['html_type'], [
-            'Multi-Select',
-            'AdvMulti-Select',
-          ])
-          ) {
+          if (is_array($fieldValue) && $this->_fields[$name][$fieldName]['html_type'] == 'CheckBox') {
```

Comment:
The history of this section of code is murky but from what I can tell, this is applying special formatting to checkboxes.
So instead of a double-negative conditional "not field types that are not checkbox" let's just say "if checkbox".
Perhaps @monishdeb can confirm this as the git history suggests he was involved writing this originally.

Reply:
+1 for this

Reply:
so it looks like possible html types are

  'CheckBox':
'Multi-Select':
 'Multi-Select State/Province':
 'Multi-Select Country':
 'Select':
 'Autocomplete-Select':
 'Radio':
 'Select Country':
 'Select State/Province':
''Select Date'
'File'
'Link' 
'Text Area'
'Text'

To  validate this we need to confirm that only the top 4 could be arrays at  this point

Reply:
@eileenmcnaughton so the two ways of writing and evaluating a refactor would be (roughly):

1. **The code analysis approach:** Analyze the original code and figure out what it does based on all possible inputs. Make sure the refactored code behaves exactly the same in all circumstances.
2. **The "what were they thinking?" approach:** Guess what the developer was trying to accomplish (which may not match what the code actually *says* because developers are only human). Make sure the refactored code satisfies their intentions.

I'm going with number 2, because based on the context (and basically the same code being duplicated in many other places) that this block is for the sake of checkboxes only, even though that's not exactly what it said.